### PR TITLE
Use delux in blinky

### PR DESCRIPTION
**Pull request by mnishiguchi**
Replace [nerves_leds](https://hexdocs.pm/nerves_leds) with [delux](https://hexdocs.pm/delux) in the `blinky` example.

Do we still need the `hello_leds` example as well?

### CODE REVIEW

**Review by jjcarstens — APPROVED** [review]
🥂 LGTM

I'm not sure if we need `hello_leds` - Need to take a closer look to see the difference, but my initial guess is "no"

**Comment by mnishiguchi** [comment]
As I quickly compare `blinky` and `hello_leds`, only remarkable difference is how blinking loop is implemented in the top-level module:

- `blinky` - loop and intervals are managed by our code
- `hello_leds` - loop and intervals are managed by `Nerves.Leds`

Other than that, it looks nearly identical to me.

```diff
❯ diff -u blinky/lib/blinky.ex  hello_leds/lib/hello_leds.ex

--- blinky/lib/blinky.ex	2022-08-16 19:56:22.000000000 -0400
+++ hello_leds/lib/hello_leds.ex	2022-08-16 19:58:29.000000000 -0400
@@ -1,9 +1,9 @@
-defmodule Blinky do
+defmodule HelloLeds do
   @moduledoc """
   Simple example to blink a list of LEDs forever.

-  The list of LEDs is platform-dependent, and defined in the config directory
-  (see config.exs). See README.md for build instructions.
+  The list of LEDs is platform-dependent, and defined in the config
+  directory (see config.exs).   See README.md for build instructions.
   """

   # Durations are in milliseconds
@@ -14,24 +14,30 @@
   require Logger

   def start(_type, _args) do
-    led_list = Application.get_env(:blinky, :led_list)
+    led_list = Application.get_env(:hello_leds, :led_list)
     Logger.debug("list of leds to blink is #{inspect(led_list)}")
-    spawn(fn -> blink_list_forever(led_list) end)
+    Enum.each(led_list, &start_blink(&1))
     {:ok, self()}
   end

-  # call blink_led on each led in the list sequence, repeating forever
-  defp blink_list_forever(led_list) do
-    Enum.each(led_list, &blink(&1))
-    blink_list_forever(led_list)
-  end
-
-  # given an led key, turn it on for @on_duration then back off
-  defp blink(led_key) do
-    # Logger.debug "blinking led #{inspect led_key}"
-    Leds.set([{led_key, true}])
-    :timer.sleep(@on_duration)
-    Leds.set([{led_key, false}])
-    :timer.sleep(@off_duration)
+  # Set led `led_key` to the state defined below. It is also possible
+  # to globally define states in `config/config.exs` by passing a list
+  # of states with the `:states` keyword.
+  #
+  # The first parameter must be an atom.
+  @spec start_blink(Keyword.T) :: true
+  defp start_blink(led_key) do
+    Logger.debug("blinking led #{inspect(led_key)}")
+    # led_key is a variable that contains an atom
+    Leds.set([
+      {
+        led_key,
+        [
+          trigger: "timer",
+          delay_off: @off_duration,
+          delay_on: @on_duration
+        ]
+      }
+    ])
   end
 end
```

Probably we no longer need `hello_leds`.

Once we know we keep one, the name of this example can be either `blinky` or `hello_leds`.

File: blinky/config/bbb.exs [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    led3: "beaglebone:green:usr3"
-  ]
+config :blinky,
+  indicators: %{
+    default: %{
+      led0: "beaglebone:green:usr0",
+      led1: "beaglebone:green:usr1",
+      led2: "beaglebone:green:usr2",
+      led3: "beaglebone:green:usr3"
+    }
```
Comment (fhunleth) [inline blinky/config/bbb.exs:11]: The Beaglebone's configuration is a bit more verbose. The idea is that each indicator could include a red, green, and blue LED. If that were the case, Delux would be able to make colors. However, it's not, so there are just 4 separate green indicators like this:

```suggestion
    default: %{
      green: "beaglebone:green:usr0",
    },
    led1: %{
      green: "beaglebone:green:usr1",
    },
    led2: %{
      green: "beaglebone:green:usr2",
    },
    led3: %{
      green: "beaglebone:green:usr3"
    }
```

File: blinky/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 The "Hello World" of the embedded world... blinking an LED
 
 * Boots to Elixir `iex` shell and blinks LEDs forever in background
-* Uses `Nerves.Leds` to manage named LEDs
+* Uses the [delux](https://hexdocs.pm/delux/readme.html) package to manage named LEDs
```
Comment (fhunleth) [inline blinky/README.md:6]:
```suggestion
* Uses the [Delux](https://hexdocs.pm/delux/readme.html) package to manage named LEDs
```

**Review by fhunleth — COMMENTED** [review]
I think the README needs a paragraph explaining what is happening. I also think that it would be nice to give some examples of what you can run at the `iex` prompt.

I'm trying to think like someone new to the project so I'm unsure about this, but I feel like the configuration of the LEDs as Delux indicators isn't obvious. I suspect people are expecting more of an Arduino-style blinking LED demo rather than one where you basically tell a library what you want and it makes it happen. Don't get me wrong. I think we're showing the right thing.

**Comment by fhunleth** [comment]
Looks great! I squashed some commits together and merged!

**Pull request merged**